### ISSUE TITLE
Error with creating Callback

### ISSUE THREAD

**Issue by icewinds** [microsoftgraph/nodejs-webhooks-rest-sample#75]
I am getting an error on auth.js - const subscriptionData = await subscriptionService.createSubscription(subscriptionConfiguration);

I am unable continue past this and my output is simply telling me [object Object]

My line in constant.js is as follows:

exports.subscriptionConfiguration = {
  changeType: 'created',
  notificationUrl: 'https://f53e028749d3.ngrok.io/listen',
  resource: 'me/events',
  clientState: 'cLIENTsTATEfORvALIDATION',
  includeResourceData: true
};

**Comment by baywet** [comment]
What permissions did you grant your application?

**Comment by icewinds** [comment]
> What permissions did you grant your application?

Thank you for the reply.

Below are the permissions that I am currently using.

![image](https://user-images.githubusercontent.com/5677428/86399171-153b7580-bca7-11ea-880b-a700aa4c7832.png)

Exchange (3)
Calendars.Read
Delegated
Read user calendars
-
Calendars.Read.All
Delegated
Read user and shared calendars
-
Calendars.Read.Shared
Delegated
Read user and shared calendars
-
Microsoft Graph (10)
Calendars.Read
Delegated
Read user calendars
-
Calendars.Read.Shared
Delegated
Read user and shared calendars
-
email
Delegated
View users' email address
-
Mail.Read
Delegated
Read user mail
-
Mail.Read.Shared
Delegated
Read user and shared mail
-
Mail.ReadBasic
Delegated
Read user basic mail
-
offline_access
Delegated
Maintain access to data you have given it access to
-
openid
Delegated
Sign users in
-
profile
Delegated
View users' basic profile
-
User.Read
Delegated
Sign in and read user profile
-

**Comment by baywet** [comment]
this looks good besides the fact you don't need the exchange permissions for that sample, only the microsoft graph ones.
As you selected delegated permissions, you signed in with your user before trying to create the subscription, correct?

**Comment by icewinds** [comment]
Yes, I have logged in with the same account on this app that I used for the AAD App Registration.

I have also updated the permissions now to below.

![image](https://user-images.githubusercontent.com/5677428/86402428-191dc680-bcac-11ea-8a42-e950e6e432f2.png)

One other issue I had was that I needed any user not just people with the same company email address to sign into this app and as far as I can see as this is based on tenant ID it has to have matching emails. is this correct?

Basically I need any one to be able to sign in with the Outlook account email and subscribe to these triggers.

Thank you

**Comment by baywet** [comment]
Can you surround the faulty line with a try catch and console log the JSON stringify representation of the error please? This way we should have a better idea of what's going on.

**Comment by icewinds** [comment]
Good afternoon, thank you for the assistance. I see now it is working. Only a few other minor issues that I have seen.

1. It is using the Beta version and this would not be good for production servers
2. I cannot extend the API to include attachments
3. Cannot get confirmation of deleted events
4. The output is repeating

Thank you so much for the time taken already.

Jason

**Comment by baywet** [comment]
Great news! for future reference, did you change anything?

It's using the beta endpoint because this sample also support change notifications with resource data which are currently in preview. But if you are only using change notifications, you can safely switch to v1.

Do you mean receiving the attachments with the change notification? If yes, this is not supported today. You can add a feature request on [uservoice](https://microsoftgraph.uservoice.com).

In your original post you were showing only subscribing to created events, you probably simply need to update the change type and recreate the subscription.

I'm not sure what you mean with the last one?

**Comment by icewinds** [comment]
Good afternoon,

inside auth.js I change the following by adding the catch function.

authRouter.get('/callback', async (req, res, next) => {
  console.log('Start Callback');
  try {
    const token = await getTokenFromCode(req.query.code);
    const subscriptionData = await createSubscription(token);
    res.redirect(
      '/home.html?subscriptionId=' + subscriptionData.id
    );
  } catch (error) {
    res.status(500);
    next(error);
  }
});

For the attachments I have updated requestHelper.js with the expand command. This brings in all the attachment data.

async getData(path) {
    const client = this.getGraphClient();
    console.log("API Path: " + path);
    const result = await client.api(path)**.expand("attachments")**.headers({
      'Content-Type': 'application/json',
      Accept: 'application/json;odata.metadata=minimal;'
                + 'odata.streaming=true;IEEE754Compatible=false'
    }).get();
    return result;
  }

For the version information. I have the below. I have tried to change beta to 1.0, v1.0 and 1 but none of those are working.

const subscription = await client.api(this.subscriptionPath).version('**beta**').create(subscriptionCreationInformation);


For the repeating items, what I mean was that once an event is triggered the result output into the browser more than once. This is something I can live with as the system is still working and I can add the required filters.

Kind regards

**Comment by baywet** [comment]
Thanks for the additional information. For the version thing, simply remove .version('beta') wherever you see it, it'll default to v1.

**Comment by icewinds** [comment]
Good evening, version worked.

Thank you
